jl4-core/src/L4/Evaluate/Value.hs

```diff
@@ -27,7 +27,7 @@ instance NFData Value where
   rnf (ValString t)               = rnf t
   rnf (ValList vs)                = rnf vs
   rnf (ValClosure given expr env) = env `seq` rnf given `seq` rnf expr
-  rnf (ValUnaryBuiltinFun r)      = rnf r
```

Comment:
doesn't this rely on the constructors of the `r` not having any lazy arguments? This could break in future without us noticing.

Reply:
I thought you would prefer it, as you wrote it 6 lines above: https://github.com/smucclaw/l4-ide/blob/cec92d08a949ebec79280a36ceb65d9d2b1067e1/jl4-core/src/L4/Evaluate/ValueLazy.hs#L89